lib/apiservers/engine/backends/container.go

```diff
@@ -1277,6 +1288,48 @@ func setPathFromImageConfig(config, imageConfig *containertypes.Config) {
 func validateCreateConfig(config *types.ContainerCreateConfig) error {
 	defer trace.End(trace.Begin("Container.validateCreateConfig"))
 
+	// process cpucount here
+	var cpuCount int64 = DefaultCPUs
+
+	// support windows client
+	if config.HostConfig.CPUCount > 0 {
+		cpuCount = config.HostConfig.CPUCount
+	} else {
+		// we hijack --cpuset-cpus in the non-windows case
+		if config.HostConfig.CpusetCpus != "" {
+			cpus := strings.Split(config.HostConfig.CpusetCpus, ",")
+			if c, err := strconv.Atoi(cpus[0]); err == nil {
+				cpuCount = int64(c)
+			} else {
+				return fmt.Errorf("Error parsing CPU count: %s", err)
+			}
+		}
+	}
+	config.HostConfig.CPUCount = cpuCount
+
+	// fix-up cpu/memory settings here
+	if cpuCount < MinCPUs {
+		config.HostConfig.CPUCount = MinCPUs
+	}
+	log.Infof("Container CPU count: %d", config.HostConfig.CPUCount)
+
+	// convert from bytes to MiB for vsphere
+	memoryMB := config.HostConfig.Memory / units.MiB
+	if memoryMB == 0 {
```

Comment:
does it mean if asked memory size is below 1MB we will set default? I would just set a minimum if not 0.